# Proyecto 4: Expresiones regulares

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import re
import os
import shutil

## Leer y guardar todos los archivos

Primero descomprimiremos el archivo zip que tiene los archivos csv

Utilizando la libreia shutil

shutil.unpack_archive('socialnetworks.zip', 'socialnetworks')

Lo que hicimos fue descomprimir el archivo con el nombre de socialnetworks y posteriormente analizar que archivos cumplen con el patron para identificar aquellos que tengan en el nombre un mes, y aquellos que no

In [2]:
ruta_zip = 'socialnetworks.zip'
ruta_destino = '/home/moni/Ciencia de datos/Proyectos/CienciaDatosLasChicasSuperpoderosas/Proyectos/Unidad2'

shutil.unpack_archive(ruta_zip,ruta_destino)
patron = r'2022'
archivos_econtrados_mes = []
archivos_econtrados_sin_mes = []

Ahora lo que haremos, sera recorrer los archivos que descomprimimos y buscamos los que tengan el año 2022

In [3]:
# Recoreremos cada carpeta

for carpeta, subcarpetas, archivos in os.walk(ruta_destino):
    for archivo in archivos:
        ruta_completa = os.path.join(carpeta, archivo)
        
        if archivo.endswith('.csv'):
            # Verificar si el archivo tiene una fecha en su nombre
            fecha_encontrada = re.search(patron, archivo)
            
            if fecha_encontrada:
                archivos_econtrados_mes.append(ruta_completa)
            else:
                archivos_econtrados_sin_mes.append(ruta_completa)

In [4]:
len(archivos_econtrados_mes)

12

In [5]:
len(archivos_econtrados_sin_mes)
archivos_econtrados_sin_mes

['/home/moni/Ciencia de datos/Proyectos/CienciaDatosLasChicasSuperpoderosas/Proyectos/Unidad2/socialnetworks/social media influencers - instagram.csv',
 '/home/moni/Ciencia de datos/Proyectos/CienciaDatosLasChicasSuperpoderosas/Proyectos/Unidad2/socialnetworks/social media influencers - youtube.csv',
 '/home/moni/Ciencia de datos/Proyectos/CienciaDatosLasChicasSuperpoderosas/Proyectos/Unidad2/socialnetworks/social media influencers - tiktok.csv']

Ya que tenemos los archivos separados en un arreglo, los podemos abrir en los dataframes que nos piden

In [6]:
Instagram = []
Patron_Insta = r'INSTAGRAM|instagram|Instagram'

Tik_tok = []
Patron_Tik_tok = r'TIKTOK|tiktok|Tiktok'

Youtube = []
Patron_Youtube = r'Youtube|youtube|YOUTUBE'

In [7]:
# Haremos una funcion que encuentre los patrones que le pasemos en unn arreglo
# Y los guarde en cierto arreglo, para que sea mas facil hacer los dataframes

def patrones(arreglo, patron, arreglo_para_guardar):
    for i in arreglo:
        r = re.search(patron, i)
        if r:
            arreglo_para_guardar.append(i)

In [8]:
patrones(archivos_econtrados_mes, Patron_Insta, Instagram)
patrones(archivos_econtrados_mes, Patron_Tik_tok, Tik_tok)
patrones(archivos_econtrados_mes, Patron_Youtube, Youtube)

## Archivos sin mes (Resumen de cada red social)

Ya que tenemos todos los archivos de las redes sociales por separado, abriremos los dataframes que no tienen fecha

In [9]:
df_Insta_s_f = pd.read_csv(archivos_econtrados_sin_mes[0])
df_Tik_tok_s_f = pd.read_csv(archivos_econtrados_sin_mes[2])
df_Youtube_s_f = pd.read_csv(archivos_econtrados_sin_mes[1])

In [10]:
df_Insta_s_f.head()

Influencer insta name instagram name          category_1 category_2  \
0                   433            433  Sports with a ball        NaN   
1          __youngbae__        TAEYANG               Music        NaN   
2           _agentgirl_  НАСТЯ ИВЛЕЕВА               Shows        NaN   
3           _imyour_joy            Joy           Lifestyle        NaN   
4         _jeongjaehyun        Jaehyun                 NaN        NaN   

  Followers Audience country(mostly) Authentic engagement\r\n  \
0     48.5M                    Spain                   383.1K   
1     12.7M                Indonesia                     478K   
2     18.8M                   Russia                   310.8K   
3     13.5M                Indonesia                     1.1M   
4     11.1M                Indonesia                     2.5M   

  Engagement avg\r\n  
0               637K  
1             542.3K  
2             377.9K  
3               1.4M  
4               3.1M

In [11]:
df_Tik_tok_s_f.head()

Tiktoker name          Tiktok name Subscribers count Views avg. Likes avg  \
0        ekin.721               MOMO’s            221.7K        26M      2.8M   
1         dojacat             Doja Cat             22.2M      25.4M        5M   
2     kiet.ac.quy           Kiệt Ắk Wỷ              2.1M      20.7M      3.5M   
3   charlidamelio      charli d’amelio            135.4M      18.7M      2.6M   
4  luvadepedreiro  Iran Ferreira (Lai)             11.4M      24.8M      2.6M   

  Comments avg. Shares avg  
0         29.4K     116.4K  
1         36.7K      46.8K  
2         38.8K      33.9K  
3         54.7K      35.2K  
4         32.7K      26.8K

In [12]:
df_Youtube_s_f.head()

youtuber name                channel name       Category Subscribers  \
0       tseries                    T-Series  Music & Dance      212.1M   
1     checkgate  Cocomelon - Nursery Rhymes      Education      132.1M   
2      setindia                   SET India            NaN      130.4M   
3     PewDiePie                   PewDiePie      Animation      111.4M   
4   MrBeast6000                     MrBeast    Video games       92.5M   

  Audience Country avg views avg likes avg comments  
0            India    323.7K      9.8K          290  
1              NaN     13.8M     80.9K          NaN  
2            India     23.6K       314           21  
3    United States      1.4M     80.8K         4.6K  
4    United States     30.6M      1.7M        67.7K

## Archivos con mes

Ahora, como se solicita que los archivos cuyo nombre tenga un mes, tambien tengan una columna que lo indique, se le agregará a cada uno

### Instagram

In [13]:
Instagram

['/home/moni/Ciencia de datos/Proyectos/CienciaDatosLasChicasSuperpoderosas/Proyectos/Unidad2/socialnetworks/social media influencers-instagram - -nov 2022.csv',
 '/home/moni/Ciencia de datos/Proyectos/CienciaDatosLasChicasSuperpoderosas/Proyectos/Unidad2/socialnetworks/social media influencers-instagram june 2022 - june 2022.csv',
 '/home/moni/Ciencia de datos/Proyectos/CienciaDatosLasChicasSuperpoderosas/Proyectos/Unidad2/socialnetworks/social media influencers-INSTAGRAM - -DEC 2022.csv',
 '/home/moni/Ciencia de datos/Proyectos/CienciaDatosLasChicasSuperpoderosas/Proyectos/Unidad2/socialnetworks/social media influencers - instagram sep-2022.csv']

In [14]:
# Archivo de Instagram - noviembre
df_I_Nov = pd.read_csv(Instagram[0])
df_I_Nov['Month'] = 'November'
df_I_Nov.head(2)

s.no         Name     Instagram Name          Category-1 Followers  \
0     1    cristiano  Cristiano Ronaldo  Sports with a ball    483.1M   
1     2  kyliejenner            Kylie 🤍             Fashion      370M   

       \nCountry Eng. (Auth.) Eng. (Avg.) Category-2     Month  
0          India         4.5M        5.6M        NaN  November  
1  United States         2.8M        4.4M   Modeling  November

In [15]:
# Archivo de Instagram - junio
df_I_June = pd.read_csv(Instagram[1])
df_I_June['Month'] = 'June'
df_I_June.head(2)

instagram name influencer name           Category_1 Category_2  \
0            433              433  Sports with a ball        NaN   
1  ____kimwoobin         김우빈           Lifestyle        NaN   

  Subscribers count Views avg. Likes avg Comments avg. Month  
0             50.7M      Spain    274.9K        456.7K  June  
1              3.6M  Indonesia    449.4K        592.7K  June

In [16]:
# Archivo de Instagram - diciembre
df_I_Dec = pd.read_csv(Instagram[2])
df_I_Dec['Month'] = 'December'
df_I_Dec.head(2)

Rank       name     instagram name          Category_1 Category_2  \
0     1   leomessi          Leo Messi  Sports with a ball     Family   
1     2  cristiano  Cristiano Ronaldo  Sports with a ball        NaN   

  followers    country Eng. (Auth.) Eng. (Avg.)     Month  
0    409.8M  Argentina        17.8M       23.4M  December  
1      523M      India        11.7M       14.7M  December

In [17]:
# Archivo de Instagram - septiembre
df_I_Sep = pd.read_csv(Instagram[3])
df_I_Sep['Month'] = 'September'
df_I_Sep.head(2)

S.no Instagram name               Name Subscribers Audience country  \
0     1      cristiano  Cristiano Ronaldo      477.9M            India   
1     2    kyliejenner            Kylie 🤍      368.1M    United States   

  Authentic engagement\n Engagement average\r\n          Category_1  \
0                     5M                   6.2M  Sports with a ball   
1                   3.5M                   5.5M             Fashion   

  Category_2      Month  
0        NaN  September  
1   Modeling  September

Una vez que cada DataFrame contiene los meses, se concatenan en uno solo y se limpia

In [18]:
# Concatenar los DataFrames de Instagram en un solo DataFrame
df_Insta = pd.concat([df_I_June, df_I_Sep, df_I_Nov, df_I_Dec], axis=0, ignore_index=True)

# Verificar el DataFrame resultante
df_Insta.head()

instagram name influencer name           Category_1   Category_2  \
0            433              433  Sports with a ball          NaN   
1  ____kimwoobin         김우빈           Lifestyle          NaN   
2   __youngbae__          TAEYANG               Music          NaN   
3    _agentgirl_    НАСТЯ ИВЛЕЕВА               Shows          NaN   
4  _hakkencoser_        Hakken 八犬           Lifestyle  Photography   

  Subscribers count     Views avg. Likes avg Comments avg. Month  S.no  ...  \
0             50.7M          Spain    274.9K        456.7K  June   NaN  ...   
1              3.6M      Indonesia    449.4K        592.7K  June   NaN  ...   
2             13.4M      Indonesia    269.6K        304.1K  June   NaN  ...   
3             18.6M         Russia    164.4K        189.4K  June   NaN  ...   
4              3.5M  United States    525.1K          643K  June   NaN  ...   

  Category-1 Followers \nCountry Eng. (Auth.) Eng. (Avg.) Category-2  Rank  \
0        NaN       NaN       NaN          NaN         NaN        NaN   NaN   
1        NaN       NaN       NaN          NaN         NaN        NaN   NaN   
2        NaN       NaN       NaN          NaN         NaN        NaN   NaN   
3        NaN       NaN       NaN          NaN         NaN        NaN   NaN   
4        NaN       NaN       NaN          NaN         NaN        NaN   NaN   

  name followers country  
0  NaN       NaN     NaN  
1  NaN       NaN     NaN  
2  NaN       NaN     NaN  
3  NaN       NaN     NaN  
4  NaN       NaN     NaN  

[5 rows x 29 columns]

In [19]:
# Limpieza del DataFrame de Instagram
    # Combinar las columnas que se refieren a la misma información y eliminar las que ya no se necesitan
#Username
df_Insta['Username'] = df_Insta['Instagram Name'].combine_first(df_Insta['instagram name']).combine_first(df_Insta['Instagram name'])
df_Insta = df_Insta.drop(['Instagram Name','instagram name','Instagram name'], axis=1)
#Name
df_Insta['Name'] = df_Insta['Name'].combine_first(df_Insta['influencer name ']).combine_first(df_Insta['name']).combine_first(df_Insta[' Name'])
df_Insta = df_Insta.drop(['influencer name ','name',' Name'], axis=1)
#Subscribers o Followers
df_Insta['Subscribers o Followers'] = df_Insta['Subscribers'].combine_first(df_Insta['Subscribers count']).combine_first(df_Insta['followers']).combine_first(df_Insta['Followers'])
df_Insta = df_Insta.drop(['Subscribers','Subscribers count','followers','Followers'], axis=1)
#Country
df_Insta['Country'] = df_Insta['Audience country'].combine_first(df_Insta['Views avg.']).combine_first(df_Insta['country']).combine_first(df_Insta['\nCountry'])
df_Insta = df_Insta.drop(['Audience country','Views avg.','country','\nCountry'], axis=1)
#Authentic
df_Insta['Authentic'] = df_Insta['Authentic engagement\n'].combine_first(df_Insta['Eng. (Auth.)']).combine_first(df_Insta['Likes avg'])
df_Insta = df_Insta.drop(['Authentic engagement\n','Eng. (Auth.)','Likes avg'], axis=1)
#Engagement
df_Insta['Engagement'] = df_Insta['Engagement average\r\n'].combine_first(df_Insta['Eng. (Avg.)']).combine_first(df_Insta['Comments avg.'])
df_Insta = df_Insta.drop(['Engagement average\r\n','Eng. (Avg.)','Comments avg.'], axis=1)
#Category1
df_Insta['Category1'] = df_Insta['Category-1'].combine_first(df_Insta['Category_1'])
df_Insta = df_Insta.drop(['Category-1','Category_1'], axis=1)
#Category2
df_Insta['Category2'] = df_Insta['Category-2'].combine_first(df_Insta['Category_2'])
df_Insta = df_Insta.drop(['Category-2','Category_2'], axis=1)
#Borrar datos inecesarios
df_Insta = df_Insta.drop(['s.no','S.no','Rank'], axis=1)

In [20]:
# Verificar el DataFrame de Instagram resultante
df_Insta

Month             Name             Username Subscribers o Followers  \
0         June              433                  433                   50.7M   
1         June         김우빈        ____kimwoobin                    3.6M   
2         June          TAEYANG         __youngbae__                   13.4M   
3         June    НАСТЯ ИВЛЕЕВА          _agentgirl_                   18.6M   
4         June        Hakken 八犬        _hakkencoser_                    3.5M   
...        ...              ...                  ...                     ...   
4067  December          thiago6     Thiago Alcantara                   10.4M   
4068  December  soycintiacossio        Cintia Cossio                      6M   
4069  December        feat.dino                   이찬                    2.7M   
4070  December       djokernole       Novak Djokovic                   11.8M   
4071  December  ravindra.jadeja  Ravindrasinh jadeja                    4.8M   

             Country Authentic Engagement           Category1  \
0              Spain    274.9K     456.7K  Sports with a ball   
1          Indonesia    449.4K     592.7K           Lifestyle   
2          Indonesia    269.6K     304.1K               Music   
3             Russia    164.4K     189.4K               Shows   
4      United States    525.1K       643K           Lifestyle   
...              ...       ...        ...                 ...   
4067  United Kingdom    129.9K     222.7K  Sports with a ball   
4068        Colombia    313.4K     390.6K            Modeling   
4069     South Korea    716.6K     876.4K           Lifestyle   
4070   United States    158.6K     197.5K  Sports with a ball   
4071           India    388.9K     485.1K           Lifestyle   

               Category2  
0                    NaN  
1                    NaN  
2                    NaN  
3                    NaN  
4            Photography  
...                  ...  
4067                 NaN  
4068       Adult content  
4069                 NaN  
4070                 NaN  
4071  Sports with a ball  

[4072 rows x 9 columns]

In [21]:
df_Insta.isna().sum()

Month                         0
Name                         47
Username                     41
Subscribers o Followers       0
Country                       9
Authentic                     0
Engagement                   16
Category1                   441
Category2                  2852
dtype: int64

In [22]:
df_Insta = df_Insta.drop_duplicates()

In [23]:
df_Insta=df_Insta.dropna()

In [24]:
df_Insta.shape

(1168, 9)

Se realizan los mismos pasos anteriores para los otros dos DataFrames de TikTok y Youtube

### TikTok

In [25]:
Tik_tok

['/home/moni/Ciencia de datos/Proyectos/CienciaDatosLasChicasSuperpoderosas/Proyectos/Unidad2/socialnetworks/social media influencers-tiktok - ---nov 2022.csv',
 '/home/moni/Ciencia de datos/Proyectos/CienciaDatosLasChicasSuperpoderosas/Proyectos/Unidad2/socialnetworks/social media influencers - Tiktok sep 2022.csv',
 '/home/moni/Ciencia de datos/Proyectos/CienciaDatosLasChicasSuperpoderosas/Proyectos/Unidad2/socialnetworks/social media influencers-tiktok june 2022 - june 2022.csv',
 '/home/moni/Ciencia de datos/Proyectos/CienciaDatosLasChicasSuperpoderosas/Proyectos/Unidad2/socialnetworks/social media influencers-TIKTOK - ---DEC 2022.csv']

In [26]:
#ARCHIVO DE NOVIEMBRE
df_T_Nov = pd.read_csv(Tik_tok[0])
df_T_Nov['Month'] = 'November'
df_T_Nov.drop(['row-cell'], axis=1, inplace=True)
df_T_Nov.head(2)

Tiktoker name Tiktok name \nFollowers Views (Avg.) Likes (Avg.)  \
0           NaN         NaN         NaN          NaN          NaN   
1   yossikramer       Yossi      145.6K        26.8M         2.2M   

  Comments (Avg.) Shares (Avg.)     Month  
0             NaN           NaN  November  
1           30.1K         71.2K  November

In [27]:
# ARCHIVO DE SEPTIEMBRE
df_T_Sep = pd.read_csv(Tik_tok[1])
df_T_Sep['Month'] = 'September'
df_T_Sep.drop(['S.no'], axis=1, inplace=True)
df_T_Sep.head(2)

Tiktoker name   Tiktok name Subscribers Views avg. Likes avg.  \
0  jypestraykids    Stray Kids       13.8M       6.4M       2.3M   
1     khaby.lame  Khabane lame      149.2M      17.3M       2.3M   

  Comments avg. Shares avg.      Month  
0         50.2K       34.2K  September  
1         15.2K        8.7K  September

In [28]:
# ARCHIVO DE JUNIO
df_T_June = pd.read_csv(Tik_tok[2])
df_T_June['Month'] = 'June'
df_T_June.head(2)

Tiktoker influencer name Subscribers count Views avg. Likes avg  \
0   ___nicoichi___   NICO【平成フラミンゴ】              1.3M       1.9M    225.3K   
1  _.anime.edits_0           video               97K       1.7M     75.6K   

  Comments avg. Shares avg Month  
0          1.7K       1.1K  June  
1          1.5K       8.7K  June

In [29]:
# ARCHIVO DE DICIEMBRE 
df_T_Dic = pd.read_csv(Tik_tok[3])
df_T_Dic['Month'] = 'December'
df_T_Dic.drop(['Rank'], axis=1, inplace=True)
df_T_Dic.head(2)

Tiktoker name Tiktok name followers views(avg) likes(avg.) comments(avg.)  \
0       mrbeast     MrBeast     60.3M      29.2M        3.5M          30.8K   
1        karolg     Karol G     42.4M      23.7M        3.4M          21.7K   

  shares(avg.)     Month  
0         7.2K  December  
1        25.7K  December

In [30]:
# Concatenar los DataFrames de TikTok en un solo DataFrame
df_Tik_tok = pd.concat([df_T_Sep, df_T_Dic, df_T_Nov, df_T_June], axis=0, ignore_index=True)

# Verificar el DataFrame resultante
df_Tik_tok.head()

Tiktoker name   Tiktok name Subscribers Views avg. Likes avg.  \
0   jypestraykids    Stray Kids       13.8M       6.4M       2.3M   
1      khaby.lame  Khabane lame      149.2M      17.3M       2.3M   
2  scarlettsspam2      scarlett        2.1M      17.9M     845.8K   
3       addisonre   Addison Rae       88.7M        22M     906.6K   
4      belindatok       Belinda        4.8M      14.2M       1.5M   

  Comments avg. Shares avg.      Month followers views(avg)  ... \nFollowers  \
0         50.2K       34.2K  September       NaN        NaN  ...         NaN   
1         15.2K        8.7K  September       NaN        NaN  ...         NaN   
2         53.9K        6.3K  September       NaN        NaN  ...         NaN   
3          7.6K       26.2K  September       NaN        NaN  ...         NaN   
4         14.5K       15.3K  September       NaN        NaN  ...         NaN   

  Views (Avg.) Likes (Avg.) Comments (Avg.) Shares (Avg.) Tiktoker  \
0          NaN          NaN             NaN           NaN      NaN   
1          NaN          NaN             NaN           NaN      NaN   
2          NaN          NaN             NaN           NaN      NaN   
3          NaN          NaN             NaN           NaN      NaN   
4          NaN          NaN             NaN           NaN      NaN   

  influencer name Subscribers count Likes avg Shares avg  
0             NaN               NaN       NaN        NaN  
1             NaN               NaN       NaN        NaN  
2             NaN               NaN       NaN        NaN  
3             NaN               NaN       NaN        NaN  
4             NaN               NaN       NaN        NaN  

[5 rows x 23 columns]

In [31]:
# Limpieza del DataFrame de Tiktok

#Username
df_Tik_tok['Username']=df_Tik_tok['Tiktoker name'].combine_first(df_Tik_tok['Tiktoker'])
df_Tik_tok.drop(['Tiktoker name', 'Tiktoker'], axis=1, inplace=True)

In [32]:
#Name
df_Tik_tok['Name']=df_Tik_tok['Tiktok name']
df_Tik_tok.drop(['Tiktok name'], axis=1, inplace=True)

In [33]:
df_Tik_tok['Name'] = df_Tik_tok['Name'].combine_first(df_Tik_tok['influencer name'])
df_Tik_tok.drop(['influencer name'], axis=1, inplace=True)

In [34]:
#Folowers o Suscribers
df_Tik_tok['Folowers o Suscribers'] = df_Tik_tok['Subscribers'].combine_first(df_Tik_tok['followers']).combine_first(df_Tik_tok['\nFollowers']).combine_first(df_Tik_tok['Subscribers count'])
df_Tik_tok.drop(['Subscribers', 'followers', '\nFollowers', 'Subscribers count'],axis=1, inplace=True)

In [35]:
#Likes
df_Tik_tok['Likes'] = df_Tik_tok['Likes (Avg.)'].combine_first(df_Tik_tok['Likes avg']).combine_first(df_Tik_tok['Likes avg.']).combine_first(df_Tik_tok['likes(avg.)'])
df_Tik_tok.drop(['Likes (Avg.)', 'Likes avg', 'Likes avg.', 'likes(avg.)'],axis=1,inplace=True)

In [36]:
#Views
df_Tik_tok['Views'] = df_Tik_tok['Views (Avg.)'].combine_first(df_Tik_tok['Views avg.']).combine_first(df_Tik_tok['views(avg)'])
df_Tik_tok.drop(['Views (Avg.)', 'Views avg.', 'views(avg)'], axis=1, inplace=True)

In [37]:
#Shares
df_Tik_tok['Shares'] = df_Tik_tok['Shares (Avg.)'].combine_first(df_Tik_tok['Shares avg']).combine_first(df_Tik_tok['Shares avg.']).combine_first(df_Tik_tok['shares(avg.)'])
df_Tik_tok.drop(['Shares (Avg.)', 'Shares avg', 'Shares avg.', 'shares(avg.)'], axis=1, inplace=True)

In [38]:
#Comments
df_Tik_tok['Comments'] = df_Tik_tok['Comments (Avg.)'].combine_first(df_Tik_tok['Comments avg.']).combine_first(df_Tik_tok['comments(avg.)'])
df_Tik_tok.drop(['Comments (Avg.)', 'Comments avg.', 'comments(avg.)'], axis=1, inplace=True)

In [39]:
# # Verificar el DataFrame de TikTok resultante
df_Tik_tok.head()

Month        Username          Name Folowers o Suscribers   Likes  \
0  September   jypestraykids    Stray Kids                 13.8M    2.3M   
1  September      khaby.lame  Khabane lame                149.2M    2.3M   
2  September  scarlettsspam2      scarlett                  2.1M  845.8K   
3  September       addisonre   Addison Rae                 88.7M  906.6K   
4  September      belindatok       Belinda                  4.8M    1.5M   

   Views Shares Comments  
0   6.4M  34.2K    50.2K  
1  17.3M   8.7K    15.2K  
2  17.9M   6.3K    53.9K  
3    22M  26.2K     7.6K  
4  14.2M  15.3K    14.5K

In [40]:
df_Tik_tok.isna().sum()

Month                    0
Username                 1
Name                     4
Folowers o Suscribers    1
Likes                    1
Views                    1
Shares                   1
Comments                 1
dtype: int64

In [41]:
df_Tik_tok.drop_duplicates()

Month          Username                Name Folowers o Suscribers  \
0     September     jypestraykids          Stray Kids                 13.8M   
1     September        khaby.lame        Khabane lame                149.2M   
2     September    scarlettsspam2            scarlett                  2.1M   
3     September         addisonre         Addison Rae                 88.7M   
4     September        belindatok             Belinda                  4.8M   
...         ...               ...                 ...                   ...   
4003       June              zeth                Zeth                 12.5M   
4004       June  zodiac.boyfriend  Zodiac Boyfriend🪐🔮                  1.8M   
4005       June       zowloficial           Lord Zowl                  1.1M   
4006       June          zrt.foot                 ZR7                100.8K   
4007       June         zubarefff   Зубарев Александр                    4M   

       Likes  Views Shares Comments  
0       2.3M   6.4M  34.2K    50.2K  
1       2.3M  17.3M   8.7K    15.2K  
2     845.8K  17.9M   6.3K    53.9K  
3     906.6K    22M  26.2K     7.6K  
4       1.5M  14.2M  15.3K    14.5K  
...      ...    ...    ...      ...  
4003    1.3M   6.7M   7.7K       3K  
4004  272.7K   1.9M   3.6K     6.8K  
4005  178.1K   1.4M   2.1K     1.4K  
4006    218K   2.3M   6.4K     1.7K  
4007    249K   1.9M   1.6K     1.3K  

[3976 rows x 8 columns]

In [42]:
df_Tik_tok=df_Tik_tok.dropna()

In [43]:
df_Tik_tok = df_Tik_tok.drop_duplicates()

In [44]:
df_Tik_tok.shape

(3972, 8)

### Youtube

In [45]:
Youtube

['/home/moni/Ciencia de datos/Proyectos/CienciaDatosLasChicasSuperpoderosas/Proyectos/Unidad2/socialnetworks/social media influencers-YOUTUBE - --DEC 2022.csv',
 '/home/moni/Ciencia de datos/Proyectos/CienciaDatosLasChicasSuperpoderosas/Proyectos/Unidad2/socialnetworks/social media influencers-youtube - --nov 2022.csv',
 '/home/moni/Ciencia de datos/Proyectos/CienciaDatosLasChicasSuperpoderosas/Proyectos/Unidad2/socialnetworks/social media influencers - Youtube sep-2022.csv',
 '/home/moni/Ciencia de datos/Proyectos/CienciaDatosLasChicasSuperpoderosas/Proyectos/Unidad2/socialnetworks/social media influencers-youtube june 2022 - june 2022.csv']

In [46]:
# Archivo de Youtube - diciembre
df_Y_Dec = pd.read_csv(Youtube[0])
df_Y_Dec['Month'] = 'December'
df_Y_Dec.head(2)

s.no Youtube channel youtuber name       Category Followers Country  \
0     1         tseries      T-Series  Music & Dance    229.5M   India   
1     2        setindia     SET India            NaN    147.2M   India   

  \nViews (Avg.) Likes (Avg.) Comments (Avg.) Category-2     Month  
0         103.8K         2.8K             158        NaN  December  
1              0            0               0        NaN  December

In [47]:
# Archivo de Youtube - noviembre
df_Y_Nov = pd.read_csv(Youtube[1])
df_Y_Nov['Month'] = 'November'
df_Y_Nov.head(2)

s.no Youtube channel               youtuber name       Category Followers  \
0     1         tseries                    T-Series  Music & Dance    229.5M   
1     2       checkgate  Cocomelon - Nursery Rhymes      Education    146.6M   

  Country \nViews (Avg.) Likes (Avg.) Comments (Avg.) Category-2     Month  
0   India         103.8K         2.8K             158        NaN  November  
1     NaN           9.5M        48.3K               0        NaN  November

In [48]:
# Archivo de Youtube - septiembre
df_Y_Sep = pd.read_csv(Youtube[2])
df_Y_Sep['Month'] = 'September'
df_Y_Sep.head(2)

S.no                        Name   Youtuber Country  Subscribers  \
0     1                    T-Series    tseries   India       224.4M   
1     2  Cocomelon - Nursery Rhymes  checkgate     NaN       142.5M   

      Category_2 Avg. views\r\n Avg. likes Avg Comments Category_3      Month  
0  Music & Dance         141.5K       4.6K          125        NaN  September  
1      Education          12.5M      64.8K            0        NaN  September

In [49]:
# Archivo de Youtube - junio
df_Y_June = pd.read_csv(Youtube[3])
df_Y_June['Month'] = 'June'
df_Y_June.head(2)

channel name  \
0                                     juptube   
1  Baby Big Mouth Kids - Sing, Dance, Explore   

                                     youTuber       Category Category_2  \
0                          Galinha Pintadinha  Music & Dance        NaN   
1  Baby Big Mouth Kids - Sing, Dance, Explore      Animation       Toys   

  Subscribers count Country Views avg. Likes avg Comments avg. Month  
0             30.5M       -          0      N/A'          N/A'  June  
1             10.6M       -      348.5         3          N/A'  June

In [50]:
# Concatenar los DataFrames de Youtube en un solo DataFrame
df_Youtube = pd.concat([df_Y_June, df_Y_Sep, df_Y_Nov, df_Y_Dec], axis=0, ignore_index=True)

# Verificar el DataFrame resultante
df_Youtube.head()

channel name  \
0                                     juptube   
1  Baby Big Mouth Kids - Sing, Dance, Explore   
2                             shfa show India   
3        Blippi - Educational Videos for Kids   
4                              infobellstamil   

                                     youTuber       Category Category_2  \
0                          Galinha Pintadinha  Music & Dance        NaN   
1  Baby Big Mouth Kids - Sing, Dance, Explore      Animation       Toys   
2                             shfa show India    Daily vlogs        NaN   
3        Blippi - Educational Videos for Kids      Education       Toys   
4                           infobells - Tamil      Education        NaN   

  Subscribers count Country Views avg. Likes avg Comments avg. Month  ...  \
0             30.5M       -          0      N/A'          N/A'  June  ...   
1             10.6M       -      348.5         3          N/A'  June  ...   
2             18.9M       -       1.2M      8.7K          N/A'  June  ...   
3             15.7M       -       1.2M      4.2K          N/A'  June  ...   
4             26.5M       -       1.3M      5.8K          N/A'  June  ...   

   Avg Comments Category_3 s.no Youtube channel youtuber name Followers  \
0           NaN        NaN  NaN             NaN           NaN       NaN   
1           NaN        NaN  NaN             NaN           NaN       NaN   
2           NaN        NaN  NaN             NaN           NaN       NaN   
3           NaN        NaN  NaN             NaN           NaN       NaN   
4           NaN        NaN  NaN             NaN           NaN       NaN   

  \nViews (Avg.) Likes (Avg.)  Comments (Avg.) Category-2  
0            NaN          NaN              NaN        NaN  
1            NaN          NaN              NaN        NaN  
2            NaN          NaN              NaN        NaN  
3            NaN          NaN              NaN        NaN  
4            NaN          NaN              NaN        NaN  

[5 rows x 26 columns]

In [51]:
# Limpieza de YouTube

# Username
df_Youtube['Username'] = df_Youtube['Youtuber'].combine_first(df_Youtube['channel name']).combine_first(df_Youtube['Youtube channel'])

In [52]:
df_Youtube.drop(['Youtuber', 'channel name','Youtube channel'],axis=1,inplace=True)

In [53]:
# Name
df_Youtube['Name'] = df_Youtube['Name'].combine_first(df_Youtube['youTuber']).combine_first(df_Youtube['youtuber name'])

In [54]:
df_Youtube.drop(['youTuber','youtuber name'],axis=1,inplace=True)

In [55]:
# Subscribers
df_Youtube['Subscribers'] = df_Youtube[' Subscribers'].combine_first(df_Youtube['Followers']).combine_first(df_Youtube['Subscribers count'])

In [56]:
df_Youtube.drop([' Subscribers', 'Followers', 'Subscribers count'],axis=1,inplace=True)

In [57]:
# Views
df_Youtube['Views'] = df_Youtube['Avg. views\r\n'].combine_first(df_Youtube['\nViews (Avg.)']).combine_first(df_Youtube['Views avg.'])

In [58]:
df_Youtube.drop(['Avg. views\r\n', '\nViews (Avg.)', 'Views avg.'],axis=1,inplace=True)

In [59]:
# Likes
df_Youtube['Likes'] = df_Youtube['Avg. likes'].combine_first(df_Youtube['Likes (Avg.)']).combine_first(df_Youtube['Likes avg'])

In [60]:
df_Youtube.drop(['Avg. likes', 'Likes (Avg.)', 'Likes avg'],axis=1,inplace=True)

In [61]:
# Comments
df_Youtube['Comments'] = df_Youtube['Avg Comments'].combine_first(df_Youtube['Comments (Avg.)']).combine_first(df_Youtube['Comments avg.'])

In [62]:
df_Youtube.drop(['Avg Comments', 'Comments (Avg.)', 'Comments avg.'],axis=1,inplace=True)

In [63]:
# Category1
df_Youtube['Category1'] = df_Youtube['Category'].combine_first(df_Youtube['Category_3'])

In [64]:
df_Youtube.drop(['Category', 'Category_3'],axis=1,inplace=True)

In [65]:
# Category2
df_Youtube['Category2'] = df_Youtube['Category_2'].combine_first(df_Youtube['Category-2'])

In [66]:
df_Youtube.drop(['Category_2', 'Category-2'],axis=1,inplace=True)

In [67]:
# Eliminacion de columnas que no queremos
df_Youtube.drop(['s.no', 'S.no'],axis=1,inplace=True)

In [68]:
df_Youtube

Country     Month                                        Name  \
0              -      June                          Galinha Pintadinha   
1              -      June  Baby Big Mouth Kids - Sing, Dance, Explore   
2              -      June                             shfa show India   
3              -      June        Blippi - Educational Videos for Kids   
4              -      June                           infobells - Tamil   
...          ...       ...                                         ...   
4066       India  December                             Classic Mr Bean   
4067  Bangladesh  December                          Anupam Movie Songs   
4068       India  December                                 Anand Audio   
4069      Russia  December                                    Wylsacom   
4070       India  December                   Eagle Home Entertainments   

                                        Username Subscribers   Views Likes  \
0                                        juptube       30.5M       0  N/A'   
1     Baby Big Mouth Kids - Sing, Dance, Explore       10.6M   348.5     3   
2                                shfa show India       18.9M    1.2M  8.7K   
3           Blippi - Educational Videos for Kids       15.7M    1.2M  4.2K   
4                                 infobellstamil       26.5M    1.3M  5.8K   
...                                          ...         ...     ...   ...   
4066                               ClassicMrBean       10.1M   48.3K   624   
4067                          Anupam Movie Songs       10.1M   41.2K   315   
4068                                  AnandAudio       10.3M  175.5K  2.2K   
4069                                    Wylsacom       10.3M  291.7K   13K   
4070                   Eagle Home Entertainments       10.2M     724     3   

     Comments             Category1        Category2  
0        N/A'         Music & Dance              NaN  
1        N/A'             Animation             Toys  
2        N/A'           Daily vlogs              NaN  
3        N/A'             Education             Toys  
4        N/A'             Education              NaN  
...       ...                   ...              ...  
4066       23                   NaN              NaN  
4067       18         Music & Dance              NaN  
4068      109                   NaN              NaN  
4069      950  Science & Technology  News & Politics  
4070        0                   NaN              NaN  

[4071 rows x 10 columns]

In [69]:
df_Youtube.isna().sum()

Country         527
Month             0
Name              0
Username          0
Subscribers       0
Views             0
Likes             0
Comments          0
Category1      1656
Category2      2555
dtype: int64

In [70]:
# Seguimos con la limpieza...
df_Youtube=df_Youtube.dropna()

In [71]:
df_Youtube = df_Youtube.drop_duplicates()

In [72]:
df_Youtube.shape

(858, 10)

## Obtención de datos de Instagram

Lo primero es obtener el número total de cuentas (Username) distintas, se sabe que el DataFrame tiene 1168 registros

In [74]:
#Muestra las cuentas duplicadas
df_Insta[df_Insta.Username.duplicated(keep=False)]

Month                 Name         Username Subscribers o Followers  \
4         June            Hakken 八犬    _hakkencoser_                    3.5M   
10        June     설인아 sᴇᴏʀɪɴᴀ         _seorina                    5.7M   
20        June       AB de Villiers   abdevilliers17                   18.3M   
33        June    Ahmed Abouhashima  ahmedabohashima                   14.8M   
49        June       Álvaro Morata     alvaromorata                   17.2M   
...        ...                  ...              ...                     ...   
4044  December     neeraj____chopra    Neeraj Chopra                    6.3M   
4049  December  officialslystallone     Sly Stallone                   15.6M   
4060  December            nickjonas       Nick Jonas                   33.5M   
4066  December             earthpix   🌎  EarthPix  🌎                   23.3M   
4068  December      soycintiacossio    Cintia Cossio                      6M   

            Country Authentic Engagement            Category1  \
4     United States    525.1K       643K            Lifestyle   
10      South Korea      728K     903.5K            Lifestyle   
20            India    274.1K     315.8K   Sports with a ball   
33            Egypt    121.9K     164.4K   Business & Careers   
49            Italy    104.7K     162.1K   Sports with a ball   
...             ...       ...        ...                  ...   
4044          India    294.7K     386.3K                Shows   
4049  United States    121.4K     153.4K          Art/Artists   
4060  United States     49.3K      70.1K                Music   
4066  United States     77.3K     100.2K  Nature & landscapes   
4068       Colombia    313.4K     390.6K             Modeling   

                      Category2  
4                   Photography  
10    Cinema & Actors/actresses  
20                    Lifestyle  
33          Finance & Economics  
49      Literature & Journalism  
...                         ...  
4044              Winter sports  
4049  Cinema & Actors/actresses  
4060                Art/Artists  
4066                Photography  
4068              Adult content  

[782 rows x 9 columns]

In [75]:
#Muestra el número total de cuentas distintas
df_Insta.Username.nunique()

775

Tambien se pide calcular el promedio de seguidores, authentic, engagement, de cada cuenta

In [76]:
# Se transforman los registros de las columnas solicitadas a valores numericos
df_Insta['Subscribers o Followers'] = df_Insta['Subscribers o Followers'].replace({'K': '*1e3', 'M': '*1e6', 'G':'*1e9'}, regex=True).map(pd.eval).astype(int)
df_Insta['Authentic'] = df_Insta['Authentic'].replace({'K': '*1e3', 'M': '*1e6', 'G':'*1e9'}, regex=True).map(pd.eval).astype(int)
df_Insta['Engagement'] = df_Insta['Engagement'].replace({'K': '*1e3', 'M': '*1e6', 'G':'*1e9'}, regex=True).map(pd.eval).astype(int)

In [77]:
df_Insta

Month               Name             Username  \
4         June          Hakken 八犬        _hakkencoser_   
10        June   설인아 sᴇᴏʀɪɴᴀ             _seorina   
11        June     SHOTARO ショウタロウ           _shotaroo_   
20        June     AB de Villiers       abdevilliers17   
33        June  Ahmed Abouhashima      ahmedabohashima   
...        ...                ...                  ...   
4064  December        labels.hybe  HYBE LABELS Updates   
4065  December          torylanez   FARGO’S CRAZY LIFE   
4066  December           earthpix       🌎  EarthPix  🌎   
4068  December    soycintiacossio        Cintia Cossio   
4071  December    ravindra.jadeja  Ravindrasinh jadeja   

      Subscribers o Followers        Country  Authentic  Engagement  \
4                     3500000  United States     525100      643000   
10                    5700000    South Korea     728000      903500   
11                    5300000      Indonesia     866900     1100000   
20                   18300000          India     274100      315800   
33                   14800000          Egypt     121900      164400   
...                       ...            ...        ...         ...   
4064                  2300000          India     891100     1000000   
4065                 11900000  United States      85100      196000   
4066                 23300000  United States      77300      100200   
4068                  6000000       Colombia     313400      390600   
4071                  4800000          India     388900      485100   

                Category1                  Category2  
4               Lifestyle                Photography  
10              Lifestyle  Cinema & Actors/actresses  
11              Lifestyle                      Music  
20     Sports with a ball                  Lifestyle  
33     Business & Careers        Finance & Economics  
...                   ...                        ...  
4064                Shows                      Music  
4065                Shows                      Music  
4066  Nature & landscapes                Photography  
4068             Modeling              Adult content  
4071            Lifestyle         Sports with a ball  

[1168 rows x 9 columns]

In [83]:
#Obtiene el promedio de las columnas solicitadas por cada cuenta
columnas = ['Subscribers o Followers','Authentic','Engagement']
Promedios = df_Insta.groupby('Username')[columnas].mean()
Promedios

Subscribers o Followers  Authentic  Engagement
Username                                                                      
! Julian                                      6450000.0   404600.0    504450.0
12 🥷🏽🖤                                        8000000.0   249150.0    402550.0
2000                                          6550000.0   420400.0    536550.0
39saku_chan                                   4300000.0   708000.0    873100.0
AB de Villiers                               19450000.0   375150.0    432750.0
...                                                 ...        ...         ...
𝐑𝐨𝐡𝐢𝐭 𝐙𝐢𝐧𝐣𝐮𝐫𝐤𝐞 ~ 𝐫𝐞𝐚𝐜𝐭𝐢𝐨𝐧𝐛𝐨𝐢 💫               24450000.0   809400.0   1100000.0
𝐓𝐎𝐇𝐈                                         12200000.0   172400.0    252750.0
𝑨𝒋𝒆𝒚 𝑵𝒂𝒈𝒂𝒓                                   16300000.0   965150.0   1138400.0
🌎  EarthPix  🌎                               23300000.0   130450.0    169200.0
📲                                             5100000.0   481450.0   1150000.0

[775 rows x 3 columns]

## Obtención de datos de TikTok

## Obtención de datos de Youtube

## ¿Cuales son las 5 cuentas mas importantes de las 3 redes sociales?